api/pkg/apis/v1alpha1/managers/secrets/secrets-manager.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package secrets
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/contexts"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/managers"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers/secret"
+	"github.com/eclipse-symphony/symphony/coa/pkg/logger"
+)
+
+var log = logger.NewLogger("coa.runtime")
+
+type SecretsManager struct {
+	managers.Manager
+	SecretProviders map[string]secret.ISecretProvider
+	Precedence      []string
+}
+
+func (s *SecretsManager) Init(context *contexts.VendorContext, cfg managers.ManagerConfig, providers map[string]providers.IProvider) error {
+	log.Debug(" M (secret): Init")
+	err := s.Manager.Init(context, cfg, providers)
+	if err != nil {
+		return err
+	}
+	s.SecretProviders = make(map[string]secret.ISecretProvider)
+	for key, provider := range providers {
+		if cProvider, ok := provider.(secret.ISecretProvider); ok {
+			s.SecretProviders[key] = cProvider
+		}
+	}
+	if val, ok := cfg.Properties["precedence"]; ok {
+		s.Precedence = strings.Split(val, ",")
+	}
+	if len(s.SecretProviders) == 0 {
+		log.Error(" M (secret): No secret providers found")
+		return v1alpha2.NewCOAError(nil, "No secret providers found", v1alpha2.BadRequest)
+	}
+	if len(s.SecretProviders) > 0 && len(s.Precedence) < len(s.SecretProviders) && len(s.SecretProviders) > 1 {
+		log.Error(" M (secret): Not enough precedence values")
+		return v1alpha2.NewCOAError(nil, "Not enough precedence values", v1alpha2.BadRequest)
+	}
+	for _, key := range s.Precedence {
+		if _, ok := s.SecretProviders[key]; !ok {
+			log.Error(" M (secret): Invalid precedence value: %s", key)
+			return v1alpha2.NewCOAError(nil, fmt.Sprintf("Invalid precedence value: %s", key), v1alpha2.BadRequest)
+		}
+	}
+	return nil
+}
+func (s *SecretsManager) Get(object string, field string, localContext interface{}) (string, error) {
```

Comment:
Can we pass the context to this function? Then the diagnostics and activity context can be passed through and logged. And please wait for log PR merged and change the logging accordingly.